core/jazz_acl/components/scm/bitbucket.js

```diff
@@ -0,0 +1,135 @@
+// =========================================================================
+// Copyright © 2017 T-Mobile USA, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// =========================================================================
+
+const request = require("request");
+const logger = require("../logger.js");
+
+const addRepoPermissionInBitbucket = async (config, serviceInfo, policies) => {
+  let users_list = [];
+  await removeAllRepoUsers(config, serviceInfo);
+  for (const policy of policies.details) {
+    try {
+      const repoInfo = {
+        "repo_name": `${serviceInfo.domain}_${serviceInfo.service}`,
+        "repo_user": policy.userId,
+        "permission": `REPO_${policy.permission}`
+      };
+      if (config.PERMISSION_CATEGORIES.includes(policy.category)) {
+        await addPermsInBitbucketRepo(config, repoInfo);
+        users_list.push(policy.userId);
+      }
+    } catch (e) {
+      logger.error("Remove user failed for user : " + user.name);
+    }
+  }
+  logger.info("Updated users: " + users_list);
+  return users_list;
+};
+
+const addPermsInBitbucketRepo = async (config, repoInfo) => {
+  try {
+    let url = `${config.HOSTNAME}${config.REPO_BASE_API}${repoInfo.repo_name}${config.REPO_USER_PERM_API}${repoInfo.permission}&name=${repoInfo.repo_user}`;
+    let payload = getPayload(config, url, 'PUT');
+    let response = await sendRequest(payload);
+    return response;
+  } catch (e) {
+    logger.error("addPermsInBitbucketRepo : " + JSON.stringify(e));
+    throw (e);
+  }
+}
+
+const removeAllRepoUsers = async (config, serviceInfo) => {
+  try {
+    let repo_name = `${serviceInfo.domain}_${serviceInfo.service}`;
+    let response = await getRepoUsers(config, repo_name);
+    let result = JSON.parse(response.body);
+    let list = [];
+    if (result.values.length > 0) {
+      for (const res of result.values) {
```

Comment:
you could have used a map here

Reply:
I am sorry. I am not clear. What is the advantage of map over here.

Reply:
eg. 
const list = result.values.map(async r => {
  await removeRepoUser(config, repo_name, user.name)
  return r.user.name;
});
await Promise.all(list);